### PR TITLE
Add netstandard support to NuGet packages

### LINKED ISSUE #171: Microsoft.Bond should support netstandard for the CoreCLR in the nuspec
Creating an issue so we can track the work here.

The nupkg should be enhanced to support the current PCL target as a `netstandard` TFM for `dotnet CLI` as a first-party so that we don't have to do the manual PCL includes from other target imports.

IE: A consumer of the project should be able to write a spec like:

``` js
{
  "version": "1.0.0-*",
  "description": "",
  "dependencies": {
    "Bond.CSharp": "4.2.0"
  },
  "frameworks": {
    "net46": {
        "dependencies": {
            "System.Runtime": "4.0.20"
        }
    },
    "netcoreapp1.0": {
      "dependencies": {
        "System.Diagnostics.Tools": "4.0.1-beta-23516",
        "System.Collections": "4.0.11-beta-23516",
        "System.Runtime": "4.0.21-beta-23516"
      }
    }
  }
}
```

Whereas today a consumer must directly import the PCL for netcore:

``` js
{
  "version": "1.0.0-*",
  "description": "",
  "dependencies": {
    "Bond.CSharp": "4.2.0"
  },
  "frameworks": {
    "net46": {
        "dependencies": {
            "System.Runtime": "4.0.20"
        }
    },
    "dotnet5.4": {
      "imports": [
        "portable-dnxcore50+net45+win8+wp8+wpa81"
      ],
      "dependencies": {
        "System.Diagnostics.Tools": "4.0.1-beta-23516",
        "System.Collections": "4.0.11-beta-23516",
        "System.Runtime": "4.0.21-beta-23516"
      }
    }
  }
}
```

### PR DESCRIPTION
**Pull request by csdahlberg**
I think this should resolve #171.

Adding support for NETStandard requires upgrading Newtonsoft.Json from 9.0.1, since that is the lowest version of that library that supports NETStandard.

Per https://docs.nuget.org/ndocs/schema/target-frameworks, the existing `Profile78` is equivalent to `netstandard1.0`, so that is what I used.

Per https://github.com/davidfowl/NetStandard/issues/5, the current way to support both PCL and NETStandard projects is to have a copy of the DLL in both lib folders in the NuGet package, so I added a `netstandard1.0` lib folder to the nuspec files that already had a `portable-net45+wp80+win8+wpa81+dnxcore50` lib folder.

All unit tests in `cs\cs.sln` and `cs\nuget\nuget.sln` passed, but I had to add an assembly binding redirect for `cs\nuget\nuget.sln` because it references the currently-published `5.0.0` NuGet packages which still reference the `7.x` version of `Newtonsoft.Json`.

I also created another test solution on my workstation with both a PCL and NETStandard project. I was able to add the modified NuGet packages to both projects, and I was able to serialize and deserialize a simple struct in CompactBinary, SimpleBinary, and Json formats.

### CODE REVIEW

**Comment by lalo** [comment]
See issues #242 and #238 for some related travis failures

File: cs/nuget/App.config [hunk 1]
```diff
@@ -0,0 +1,12 @@
+﻿<?xml version="1.0" encoding="utf-8" ?>
+<configuration>
+  <runtime>
+    <assemblyBinding xmlns="urn:schemas-microsoft-com:asm.v1">
+      <dependentAssembly>
+        <assemblyIdentity name="Newtonsoft.Json" publicKeyToken="30ad4fe6b2a6aeed" culture="neutral" />
+        <!-- Required until the published Bond libraries have a Newtonsoft.Json dependency of v9.0.1 -->
```
Comment (chwarr) [inline cs/nuget/App.config]: Makes sense.

**Comment by csdahlberg** [comment]
I signed the CLA on the 18th, and got an email saying it was approved by the CLA admin on the 19th.

**Comment by chwarr** [comment]
Closing to re-open again to kick the CLA bot.

**Comment by chwarr** [comment]
Re-opening to re-run the CLA bot.

**Comment by chwarr** [comment]
I haven't forgotten about this. I'm in the middle of some other higher priority work at the moment. I expect to be able to look at this on Wednesday.

**Comment by chwarr** [comment]
I was able to successfully test this with our build and release scripting. I also got a simple application to run on Ubuntu with .NET Core 1.0.0-preview2-003131.

This seems like enough to get started with a preview release while I'm also working on getting Bond to build directly with dotnet.

I'll need to wait on merging this for a bit. There's a large change coming in to master in the next couple of days.

**Comment by Petermarcu** [comment]
Great to see Bond supporting .NET Standard! Keep up the great work! :)

**Comment by chwarr** [comment]
There are a few build issues we need to fix after the big merge that just came in. Once we've fixed those, I'll be able to merge this in.

**Comment by chwarr** [comment]
I'm running the [merge](https://github.com/Microsoft/bond/tree/merge-ns) through the CI builds now. I resolved the conflict mentioned, and updated the CHANGELOG as well.

* [Travis](https://travis-ci.org/Microsoft/bond/builds/174085828)
* [AppVeyor](https://ci.appveyor.com/project/sapek/bond/build/1.0.1004)

We expect the doc and cpp-clang builds to fail at the moment.

**Comment by chwarr** [comment]
FYI: the version just released, 5.1.0, includes .NET Core support via .NET Standard 1.0, 1.3, or 1.6. This incorporates and extends @csdahlberg's work.

**Pull request merged**